lib/spack/spack/solver/asp.py

```diff
@@ -1104,7 +1104,13 @@ def preferred_variants(self, pkg_name):
             if not isinstance(values, tuple):
                 values = (values,)
 
+            spec = spack.spec.Spec(pkg_name)
+            spec.add_variant_values(variant_name,
+                                    values)
```

Comment:
Can you add a comment explaining that we make this call to validate the variant? It was not obvious to me, without looking at the side effects of `add_variant_values` to understand why we would construct a spec that is never used in the function.

Also, minor formatting request:
```suggestion
            spec = spack.spec.Spec(pkg_name)
            spec.add_variant_values(variant_name, values)
```